proposals/2024/10-01_projektor_integration.md

```diff
@@ -0,0 +1,232 @@
+  - name: test-report
+    image: vela/test-report-agent:latest
+    parameters:
+      report_format: json
+      report_path: "./test-results.json"
+      retention_days: 30
+      notify_slack: true
+```
+In this example:
+- **`report_format`** specifies the format of the test results (e.g., JSON, JUnit XML).
+- **`report_path`** defines the path to the test results file generated in the previous steps.
+- **`retention_days`** allows users to set a retention policy for test data.
+- **`notify_slack`** indicates whether to send notifications to Slack regarding the test results.
+### 2. Test Report Step Workflow
+- **Execution**:
+    - The pipeline’s test steps execute as usual, generating a test results file (e.g., `test-results.json`).
+    - The `test-report` step runs afterward, using the `vela/test-report-agent` image to parse and submit test results to the Vela backend.
+### 3. Backend Enhancements
+To support this new feature, Vela’s backend will require additional API endpoints and an expanded database schema.
+#### Proposed Database Tables for Vela's Test Reporting
+Based on the Projektor.dev architecture, here is a comprehensive list of tables required to store and manage test results, code quality, and related metrics in Vela's backend:
+1. **`code_coverage_file`**: Stores file-level details for code coverage.
+2. **`code_coverage_group`**: Represents groups or categories of coverage data.
+3. **`code_coverage_stats`**: Holds detailed coverage statistics like statements covered, lines missed, and branches missed.
+4. **`code_coverage_run`**: Aggregates coverage data for a specific run.
+5. **`performance_results`**: Contains performance metrics, such as request count, average time, and maximum response time.
+6. **`code_quality_report`**: Stores code quality report data, including file and group names.
+7. **`test_run`**: Represents a single test run, storing counts of passed, failed, and skipped tests along with timing details.
+8. **`results_metadata`**: Contains metadata related to each test run, such as CI information and group labels.
+9. **`test_run_system_attributes`**: Stores system attributes for test runs, such as pinned status.
+10. **`test_suite`**: Represents high-level test groupings, like test suites, with success and failure counts.
+11. **`test_suite_group`**: Groups related test suites together for organized reporting.
+12. **`test_case`**: Stores individual test cases, including their results, duration, and logs.
+13. **`test_failure`**: Captures detailed information on test failures, including failure messages and stack traces.
+14. **`test_run_attachment`**: Manages attachments for a test run, like log files and screenshots.
+15. **`results_processing`**: Logs the status and errors (if any) from test results processing.
+16. **`results_processing_failure`**: Tracks specific failure cases encountered during test processing.
+17. **`shedlock`**: Used for distributed locking to prevent concurrent processing issues.
```

Comment:
curious if there's opportunity for consolidation? i know it's modeled after projektor (minus tables we can leverage today), but this adds as many tables as we have currently (roughly).